filebeat/input/runnerfactory.go

```diff
@@ -58,10 +58,14 @@ func (r *RunnerFactory) Create(
 }
 
 func (r *RunnerFactory) CheckConfig(cfg *common.Config) error {
-	_, err := r.Create(pipeline.NewNilPipeline(), cfg)
+	runner, err := r.Create(pipeline.NewNilPipeline(), cfg)
 	if _, ok := err.(*common.ErrInputNotFinished); ok {
 		// error is related to state, and hence config can be considered valid
 		return nil
 	}
-	return err
+	if err != nil {
+		return err
+	}
+	runner.Stop()
```

Comment:
`Stop` is not blocking I think. Or is it? We also have a `Wait` method on some inputs. If we do not 'block' there might be a chance that our tests become flaky due to the go-routine check and resources being freed eventually asynchronously.

Reply:
Actually this stops the runner. `Wait` is not exposed in the runner.

The goroutine checker has a wait loop with a configurable timeout, we already have non-flaky tests that stop the inputs and wait for their goroutines to finish. I wouldn't expect this to introduce additional flakiness.